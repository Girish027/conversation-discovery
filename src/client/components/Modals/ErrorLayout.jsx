import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { AlertIcon } from '@tfs/ui-components';
import { modalErrorLayout } from '../../styles/index';

export default class ErrorLayout extends PureComponent {
  render() {
    const { errorMsg, styleOverride } = this.props;

    const layoutStyle = Object.assign({}, modalErrorLayout, styleOverride);

    return (
      <div
        style={layoutStyle}
      >
        <AlertIcon />
        {' '}
        {errorMsg}
      </div>
    );
  }
}

ErrorLayout.propTypes = {
  errorMsg: PropTypes.string.isRequired,
  styleOverride: PropTypes.object
};

ErrorLayout.defaultProps = {
  styleOverride: {}
};
