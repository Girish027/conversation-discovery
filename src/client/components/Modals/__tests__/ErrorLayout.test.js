import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ErrorLayout from '../ErrorLayout';

describe('<ErrorLayout />', function () {
  let wrapper;


  const props = {
    errorMsg: 'Project ABC already exist',
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<ErrorLayout
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with default props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when styleOverride props is passed', () => {
      const newProps = {
        errorMsg: 'Run ABC already exist',
        styleOverride: { marginBottom: '20px' }
      };
      wrapper = getShallowWrapper(newProps);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
