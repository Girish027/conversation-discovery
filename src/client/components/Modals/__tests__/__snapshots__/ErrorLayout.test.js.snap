// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorLayout /> Snapshots renders correctly when styleOverride props is passed 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFFFD1",
      "border": "1px solid #C81919",
      "borderRadius": "3px",
      "color": "#C81919",
      "fontSize": "15px",
      "marginBottom": "20px",
      "padding": "10px",
    }
  }
>
  <AlertIcon
    exclamationMarkColor="#FFF"
    fill="#FF3C27"
    height="1em"
    width="1em"
  />
   
  Run ABC already exist
</div>
`;

exports[`<ErrorLayout /> Snapshots renders correctly with default props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FFFFD1",
      "border": "1px solid #C81919",
      "borderRadius": "3px",
      "color": "#C81919",
      "fontSize": "15px",
      "padding": "10px",
    }
  }
>
  <AlertIcon
    exclamationMarkColor="#FFF"
    fill="#FF3C27"
    height="1em"
    width="1em"
  />
   
  Project ABC already exist
</div>
`;
